ports/vcpkg-python-scripts/vcpkg_python_functions.cmake

```diff
@@ -82,8 +80,16 @@ function(vcpkg_python_build_wheel)
     endforeach()
   endif()
 
+  vcpkg_cmake_get_vars(cmake_vars_file)
+  include("${cmake_vars_file}")
+
+  set(ENV{CFLAGS} "${VCPKG_COMBINED_C_FLAGS_RELEASE}")
+  set(ENV{CXXFLAGS} "${VCPKG_COMBINED_CXX_FLAGS_RELEASE}")
+  set(ENV{LDFLAGS} "${VCPKG_COMBINED_SHARED_LINKER_FLAGS_RELEASE}")
+  set(ENV{ARFLAGS} "${VCPKG_COMBINED_STATIC_LINKER_FLAGS_RELEASE}")
+
   vcpkg_execute_required_process(
-    COMMAND ${env} "${z_vcpkg_python_func_python}" -m gpep517 build-wheel --wheel-dir "${z_vcpkg_wheeldir}" --output-fd 1 ${build_ops}
+    COMMAND ${env} "${PYTHON3}" -m gpep517 build-wheel --wheel-dir "${z_vcpkg_wheeldir}" --output-fd 1 ${build_ops}
```

Comment:
@Neumann-A what was your intent with `z_vcpkg_python_func_python`? Should this be the system python or something provided through vcpkg?

In the current state of this repository, it happens to be system python. In this case I think we'll need to acquire build dependencies (e.g. `gpep517`) similar to the `py-meson` approach above
https://github.com/open-vcpkg/python-registry/pull/51/files#diff-84817e4d8adab62c915c43e6a931b961f018d05eca77fc90b42fc62d032e8298R27

Reply:
The idea was to have a single place which handles which python gets used instead of having to change all vcpkg_execute_required_process calls

Reply:
where would this ideally be defined? in vcpkg-python-scripts/vcpkg-port-config.cmake ?

Reply:
I think it doesn't matter if the port-config or the scripts defined them. Thy will be in close proximity any way.

Reply:
- `PYTHON3` is set by `vcpkg-tool-meson/vcpkg-port-config.cmake`
- we need a python with build deps installed (e.g. `gpep517`)
- if we use `x_vcpkg_get_python_packages` to acquire missing packages, we need to replace `PYTHON3` with a python from the venv with the packages installed
- build deps are known by the consuming port (`py-matplotlib`) in this case, but I don't see how this could replace `PYTHON3` early enough

Options:

a) Use python from a host vcpkg port and specify build deps in `vcpkg.json`
b) Make it possible to inject build deps into `vcpkg-python-scripts` and create a venv in there